api/v1/db/oracledb/parks-dao.js

```diff
@@ -0,0 +1,188 @@
+  const sqlParams = {
+    parkId: id,
+  };
+  const queryParams = `
+    AND ID = :parkId
+  `;
+  const sqlQuery = `${sql}${queryParams}`;
+  const connection = await conn.getConnection();
+  try {
+    const { rows } = await connection.execute(sqlQuery, sqlParams);
+    if (_.isEmpty(rows)) {
+      return undefined;
+    }
+    if (rows.length > 1) {
+      throw new Error('Expect a single object but got multiple results.');
+    } else {
+      const serializedPark = serializePark(rows[0]);
+      return serializedPark;
+    }
+  } finally {
+    connection.close();
+  }
+};
+
+/**
+ * @summary Post parks
+ * @param {object} parkBody park body object
+ * @returns {Promise<object>} Promise object represents a specific park or return undefined if term
+ *                            is not found
+ */
+const postParks = async (parkBody) => {
+  const { attributes, relationships } = parkBody.data;
+  const { amenities, location } = attributes;
+  const sqlBinds = {
+    name: attributes.name,
+    streetAddress: location.streetAddress,
+    city: location.city,
+    state: location.state,
+    zip: location.zip,
+    latitude: location.latitude,
+    longitude: location.longitude,
+    ownerId: relationships.owner.data.id,
+    outId: { type: oracledb.NUMBER, dir: oracledb.BIND_OUT },
+  };
+  const sqlQuery = `
+    INSERT INTO PARKS (NAME, STREET_ADDRESS, CITY, STATE, ZIP, LATITUDE, LONGITUDE, OWNER_ID,
+    BALLFIELD, BARBEQUE_GRILLS, BASKETBALL_COURTS, BIKE_PATHS, BOAT_RAMPS, DOGS_ALLOWED,
+    DRINKING_WATER, FISHING, HIKING_TRAILS, HORSESHOES, NATURAL_AREA, OFFLEASH_DOG_PARK,
+    OPEN_FIELDS, PICNIC_SHELTERS, PICNIC_TABLES, PLAY_AREA, RESTROOMS, SCENIC_VIEW_POINT,
+    SOCCER_FIELDS, TENNIS_COURTS, VOLLEYBALL) 
+    VALUES (
+      :name, :streetAddress, :city, :state, :zip, :latitude, :longitude, :ownerId,
+      ${amenities.includes('ballfield') ? 1 : 0},
+      ${amenities.includes('barbequeGrills') ? 1 : 0},
+      ${amenities.includes('basketballCourts') ? 1 : 0},
+      ${amenities.includes('bikePaths') ? 1 : 0},
+      ${amenities.includes('boatRamps') ? 1 : 0},
+      ${amenities.includes('dogsAllowed') ? 1 : 0},
+      ${amenities.includes('drinkingWater') ? 1 : 0},
+      ${amenities.includes('fishing') ? 1 : 0},
+      ${amenities.includes('hikingTrails') ? 1 : 0},
+      ${amenities.includes('horseshoes') ? 1 : 0},
+      ${amenities.includes('naturalArea') ? 1 : 0},
+      ${amenities.includes('offleashDogPark') ? 1 : 0},
+      ${amenities.includes('openFields') ? 1 : 0},
+      ${amenities.includes('picnicShelters') ? 1 : 0},
+      ${amenities.includes('picnicTables') ? 1 : 0},
+      ${amenities.includes('playArea') ? 1 : 0},
+      ${amenities.includes('restrooms') ? 1 : 0},
+      ${amenities.includes('scenicViewPoint') ? 1 : 0},
+      ${amenities.includes('soccerFields') ? 1 : 0},
+      ${amenities.includes('tennisCourts') ? 1 : 0},
+      ${amenities.includes('volleyball') ? 1 : 0}
```

Comment:
I think you should be able to refactor this logic using `_.reduce()`. Also, you should be able to use the enums from your openapi file here.